components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/handlers/grant/AbstractAuthorizationGrantHandler.java

```diff
@@ -804,6 +824,14 @@ private void updateRefreshTokenValidityPeriodInMessageContext(OAuthAppDO oAuthAp
                         "validity period configured from OAuthTokenReqMessageContext: " +
                         refreshTokenValidityPeriodInMillis + " ms");
             }
+        } else if (existingTokenBean != null && !isTokenRenewalPerRequestConfigured() &&
+                isRefreshTokenValid(existingTokenBean, tokReqMsgCtx.getValidityPeriod(), tokenReq.getClientId()) &&
+                !isExtendedToken) {
+            /*
+             Checking whether existing refresh token will be reused, if so the refresh expiry time
+             should be reused instead of calculating the refresh expiry from the app level.
+             */
+            refreshTokenValidityPeriodInMillis = existingTokenBean.getRefreshTokenValidityPeriodInMillis();
```

Comment:
What's the reason to include this new logic here

Reply:
As we discussed offline, the reason for adding this logic will be re-evaluated based on the extension timeout and will be fixed via a separate defect.